test/e2e/test/run_mutation.go

```diff
@@ -29,8 +29,9 @@ func RunMutations(t *testing.T, creationBuilders []Builder, mutationBuilders []B
 		steps = steps.WithSteps(mutateTo.MutationTestSteps(k))
 	}
 
-	for _, mutateTo := range mutationBuilders {
-		steps = steps.WithSteps(mutateTo.DeletionTestSteps(k))
+	// Delete using the original builder (so that we can use it as a mutation builder as well)
```

Comment:
Not sure I understand what difference that makes?

Reply:
My hacky way of testing full configuration reversal is to specify have the initial builder as a second mutation builder:
`initial config -> mutated config -> initial config` 

If we delete based on the mutation builders the tests fail because it will run the deletion steps for both mutation builders: `mutated config` and `initial config` both using the same cluster, the second attempt to delete the same cluster will obviously fail.

Given that the identity of the cluster has to stay the same between creation and mutation deleting based on the original builder is fine. And we can now run (almost) infinite combinations of builders for the same cluster as mutation steps.

Reply:
Got it 👍